plugin/advisor/catalog/type.go

```diff
@@ -0,0 +1,87 @@
+package catalog
+
+type stateInt struct {
+	defined bool
```

Comment:
valid? 

to be consistent with golang sql package

Reply:
https://sourcegraph.com/github.com/golang/go/-/blob/src/database/sql/sql.go?L213

In the `sql` package, there are types like
```go
type NullInt64 struct {
	Int64 int64
	Valid bool // Valid is true if Int64 is not NULL
}
```
which seems similar.